### PR TITLE
feat(ai): tags

### PR DESCRIPTION
**Pull request by vojtaholik**
screenshots

gif

### CODE REVIEW

File: apps/ai-hero/src/app/(content)/posts/_components/edit-post-form-metadata.tsx [hunk 1]
```diff
@@ -3,10 +3,13 @@ import { Suspense, use } from 'react'
 import { useRouter } from 'next/navigation'
 import { PostPlayer } from '@/app/(content)/posts/_components/post-player'
 import { reprocessTranscript } from '@/app/(content)/posts/[slug]/edit/actions'
+import AdvancedTagSelector from '@/components/resources-crud/tag-selector'
```
Comment (joelhooks) [inline apps/ai-hero/src/app/(content)/posts/_components/edit-post-form-metadata.tsx:6]: this should probably be promoted to package level

File: apps/ai-hero/src/app/admin/tags/page.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+export default async function TagManagementPage() {
+	const { ability } = await getServerAuthSession()
+
+	if (!ability.can('manage', 'all')) {
+		notFound()
+	}
+
+	const tags = await getTags()
+	return <TagManagement initialTags={tags} />
```
Comment (joelhooks) [inline apps/ai-hero/src/app/admin/tags/page.tsx:15]: the ability to add tags "on the fly" while posting is probably sensible for a future PR

File: apps/ai-hero/src/lib/tags-query.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import { db } from '@/db'
+import { tag as tagTable } from '@/db/schema'
+import { eq } from 'drizzle-orm'
+import { z } from 'zod'
+
+import { Tag, TagSchema } from './tags'
+
+export async function getTags(): Promise<Tag[]> {
+	const tags = await db.query.tag.findMany()
+	console.log({ tags })
```
Comment (joelhooks) [inline apps/ai-hero/src/lib/tags-query.ts:13]:
```suggestion
```

File: apps/ai-hero/src/lib/tags-query.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+	return z.array(TagSchema).parse(tags)
+}
+
+type CreateTagInput = z.infer<typeof TagSchema>
+
+export async function createTag(input: CreateTagInput): Promise<Tag> {
+	console.log('creating tag', input)
```
Comment (joelhooks) [inline apps/ai-hero/src/lib/tags-query.ts:20]:
```suggestion
```

File: apps/ai-hero/src/lib/tags-query.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+	const validatedInput = TagSchema.parse(input)
+	console.log({ validatedInput })
```
Comment (joelhooks) [inline apps/ai-hero/src/lib/tags-query.ts:22]:
```suggestion
```

File: apps/ai-hero/src/app/(content)/posts/_components/edit-post-form-metadata.tsx (hunk 1 above)
Comment (vojtaholik) [reply apps/ai-hero/src/app/(content)/posts/_components/edit-post-form-metadata.tsx]: moved to ui package 👍

**Review by joelhooks — APPROVED** [review]

**Pull request merged**